### PR TITLE
[Backport release-23.11] zfs_2_2: 2.2.3 -> 2.2.4; zfs_unstable: 2.2.3-unstable-2024-04-09 -> 2.2.4

### PR DESCRIPTION
**Pull request by numinit**
## Description of changes

Manual backport of #308642 to 23.11 since Linux 6.8 was backported too.

## Things done

`nix-build zfsUnstable.passthru.tests -A zfs.passthru.tests`

`nix-build -A python3Packages.py-libzfs`

More tests pending.

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/py-libzfs/default.nix [hunk 1]
```diff
@@ -2,21 +2,30 @@
...
   };
 
+  patches = [
+    (fetchpatch2 {
+      url = "https://github.com/truenas/py-libzfs/commit/b5ffe1f1d6097df6e2f5cc6dd3c968872ec60804.patch";
+      hash = "sha256-6r5hQ/o7c4vq4Tfh0l1WbeK3AuPvi+1wzkwkIn1qEes=";
+    })
+  ];
+
   nativeBuildInputs = [ cython ];
```
Comment (numinit) [inline pkgs/development/python-modules/py-libzfs/default.nix:29]: main difference was cython vs cython_0

**Review by crertel — APPROVED** [review]

**Review by nicoonoclaste — APPROVED** [review]

**Comment by risicle** [comment]
Not a show stopper, but you have rather managed to muddle up the contents of the commits compared to cherry-pick references, e.g. moved the fetchpatch addition from one commit to another.

**Comment by numinit** [comment]
Thanks, I think I know what happened here.

**Review by crertel — APPROVED** [review]

**Comment by numinit** [comment]
I just used:

`git cherry-pick -x c6a3a06e5dec34aaaf80576760a81973644e3864 2cbe0633194e987048207cd79a662ce761f56457 a1dcb54257af49396329dc8467a465b21d7746b9`

And then fixed the patches.

I wonder if it's because it signed the merge commits with my GPG key afterwards...

**Comment by adamcstephens** [comment]
This is a big jump for the python library. Is this necessary for the zfs version updates?

**Comment by numinit** [comment]
I thought about excluding it, but https://github.com/NixOS/nixpkgs/pull/308642#issuecomment-2092844062 seemed to imply it was necessary.

**Comment by numinit** [comment]
Let me give that a shot then. :-)

**Comment by numinit** [comment]
It looks like it applies and builds fine. Re-running the NixOS tests just to be sure.

**Comment by numinit** [comment]
All good. Pushed it up without the GPG sigs, we'll see if that works this time.

**Comment by adamcstephens** [comment]
@ofborg build zfsUnstable.passthru.tests zfs.passthru.tests

**Pull request merged**